src/screens/CameraScreen.js

```diff
@@ -1,26 +1,296 @@
-import React from "react";
-import { View, Text, StyleSheet, Button } from "react-native";
-
-const CameraScreen = (props) => {
-  return (
-    <View style={styles.screen}>
-      <Text>Camera Screen</Text>
-      <Button
-        title="Posting"
-        onPress={() => {
-          props.navigation.navigate({ routeName: "Posting" });
-        }}
-      />
-    </View>
-  );
-};
+import React, { Component } from "react";
+import { View, Text, SafeAreaView, StatusBar, StyleSheet } from "react-native";
+import { MaterialIcons } from "@expo/vector-icons";
+import { Camera } from "expo-camera";
+import * as Permissions from "expo-permissions";
+import { Video } from "expo-av";
+import { TouchableOpacity } from "react-native-gesture-handler";
+import Colors from "../constants/Colors";
+
+export default class CameraScreen extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hasCameraPermissions: null,
+      type: Camera.Constants.Type.back,
+      flash: Camera.Constants.FlashMode.off,
+      videoTaken: false,
+      cameraIsRecording: false,
+      video: null,
+    };
+  }
+
+  componentDidMount = async () => {
+    const camera = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({
+      hasCameraPermissions: camera.status === "granted",
```

Comment:
Can we have statuses like "granted" as a constant? If there are other statuses like this we use throughout the codebase we could create a new file in the constants directory.

Reply:
Oh perhaps there's a constant within `Camera.Constants`?

Reply:
Hmm not sure if that's necessary? Show me how lol

Reply:
Not a blocker, let's keep it as is :)